setup.py

```diff
@@ -52,13 +52,24 @@ def build_extension(self, ext: CMakeExtension):
 
         # Set Python_EXECUTABLE instead if you use PYBIND11_FINDPYTHON
         configure_args = [
-            "-GNinja",
             f"-DCMAKE_LIBRARY_OUTPUT_DIRECTORY={extdir}",
             f"-DPYTHON_EXECUTABLE={sys.executable}",
-            f"-DCMAKE_MAKE_PROGRAM={ninja_path}",
             "-DENABLE_WARNINGS=OFF",  # Ignore warnings
         ]
 
+        if platform.system() == "Windows":
+            # As Ninja does not support long path for windows yet:
+            #  (https://github.com/ninja-build/ninja/pull/2056)
+            configure_args += [
+                "-G Visual Studio 17 2022", 
```

Comment:
What does this line do? Does it require an explicit version of visual studio to work?

Reply:
This is the version we use to build the wheels. It may not be necessary, but it seems to be common practice to set the Visual Studio generator before the clang-cl toolchain.

Reply:
Is there a better way to programmatically determine or set this? I don't think we should hardcode the version in here.

Reply:
I will try without it and see if this will default to the right generator.

Reply:
Done!